sdks/java/core/src/main/java/org/apache/beam/sdk/options/ManualDockerEnvironmentOptions.java

```diff
@@ -32,6 +32,14 @@
 
   void setRetainDockerContainers(boolean retainDockerContainers);
 
+  @Description(
+      "Mount temporary directory to Docker container instance. "
+          + "The parameter is a path relative to the system temporary directory "
+          + "e.g. '--mountTempDir foo' implies '/tmp/foo' on Linux.")
+  String getMountTempDir();
```

Comment:
Lets also specify where the mount will be available in docker. 
We can standardize the name path in docker to `/tmp/beam_mount/<mount_dir>`
example `/tmp/beam_mount/tmp/foo`

Reply:
Wouldn't it be better to keep paths as close as possible like /tmp/foo (local) and /tmp/foo (docker)? It might be confusing if we mount /tmp/foo (local) to /tmp/beam_mount/tmp/foo (docker).

Reply:
What about conflicts (e.g. if the directory already exists in the docker image). Also, should we allow specifying multiple mount dirs?

Reply:
We can put a unique prefix for helping avoid possible conflict but I think still a rule of thumb is that users need to be cautious in choosing mounting directory name. In case of this new option, the base directory is fixed to /tmp so conflicts might not easily happen.

For multiple mount dirs, I'm not sure how much it's useful since users can create as many directories as they want in a single mounting directory.